cloudinary-core/src/main/java/com/cloudinary/Api.java

```diff
@@ -789,6 +790,47 @@ public ApiResponse reorderMetadataFields(String orderBy, String direction, Map o
         return callApi(HttpMethod.PUT, uri, map, options);
     }
 
+    public ApiResponse listMetadataRules(Map options) throws Exception {
+        if (options == null || options.isEmpty()) options = ObjectUtils.asMap();
+        final Map params = new HashMap();
+        List<String> uri = Arrays.asList("metadata_rules");
+        return callApi(HttpMethod.GET, uri, params, options);
+    }
+
+    public ApiResponse addMetadataRule(MetadataRule rule, Map options) throws Exception {
+        if (options == null || options.isEmpty()) options = ObjectUtils.asMap();
+        options.put("content_type", "json");
+        final Map params = new HashMap();
```

Comment:
why not to implement `asMap` inside `MetadataRule` and put all this logic there?